### PR TITLE
fix(extension): address feedly bug

### PR DESCRIPTION
**Pull request by cyndichin**
## Summary
Fix bug where we cannot save from Feedly due to URL initialization returning nil

## References 
* IN-1254

## Implementation Details
Create a helper function to parse a URL from a string. Modified from https://www.hackingwithswift.com/example-code/strings/how-to-detect-a-url-in-a-string-using-nsdatadetector

This only applied to the items that are of the type `public.plain-text`

## Test Steps
Nav to the Feedly app
Share any item to Pocket
Verify that error no longer appears and you are able to save the item

## PR Checklist:
- [x] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/SaveToPocketKit/SavedItem/SavedItemViewModel.swift [hunk 1]
```diff
@@ -143,7 +143,7 @@ extension SavedItemViewModel {
 
             if provider.hasItemConformingToTypeIdentifier(plainTextUTI) {
                 guard let string = try? await provider.loadItem(forTypeIdentifier: plainTextUTI, options: nil) as? String,
-                      let url = URL(string: string) else {
```
Comment (cyndichin) [inline PocketKit/Sources/SaveToPocketKit/SavedItem/SavedItemViewModel.swift:146]: We were receiving `"The Continentals first trailer takes the John Wick series to the 70s https://www.theverge.com/23680378/the-continental-teaser-trailer-john-wick-peacock"` as the string and so this value returned nil

**Review by bassrock — APPROVED** [review]

**Comment by bassrock** [comment]
@cyndichin can you also UI test this?

**Comment by cyndichin** [comment]
> @cyndichin can you also UI test this?

I can possible hack one if I figure out how to type in the Reminders app. 
1. Create a reminder with the text "Get Pocket get pocket.com"
2. Highlight that text
3. Navigate to Share setting
4. Verify it has been saved

I was unable to find an app on Simulator that shared in the same format. 
Do you have a better alternative and worth investing time into the steps above?

**Comment by bassrock** [comment]
Ah i see. thats ok then i didnt think about what we needed to simulate this in ui.

**Comment by cyndichin** [comment]
> Ah i see. thats ok then i didnt think about what we needed to simulate this in ui.

Oh I just saw this, let me know if you think we should keep the tests or not. It passes locally.

**Comment by bassrock** [comment]
@cyndichin if the tests work lets keep it! I think they are important

**Pull request merged**